dotCMS/src/main/java/com/dotcms/filters/interceptor/saml/SamlWebInterceptor.java

```diff
@@ -0,0 +1,529 @@
+
+                Logger.debug(this, ()-> "- Logging out through the dotCMS Login Service...");
+                this.loginService.doLogout(request, response);
+            }
+        }
+    }
+
+    public void doRequestLoginSecurityLog(final HttpServletRequest request,
+                                          final IdentityProviderConfiguration identityProviderConfiguration) {
+
+        try {
+
+            final Host host  = this.hostWebAPI.getCurrentHost(request);
+            final String env = this.samlWebUtils.isFrontEndLoginPage(request.getRequestURI()) ? "frontend" : "backend";
+            final String log = new Date() + ": SAML login request for Site '" + host.getHostname() + "' with IdP ID: "
+                    + identityProviderConfiguration.getId() + " (" + env + ") from " + request.getRemoteAddr();
+
+            // “$TIMEDATE: SAML login request for $host (frontend|backend) from  $REQUEST_ADDR”
+            SecurityLogger.logInfo(SecurityLogger.class, this.getClass()  + " - " + log);
+            Logger.debug(this, log);
+        } catch (Exception e) {
+            Logger.error(this, e.getMessage(), e);
+        }
+    }
+
+    protected void doAuthenticationLoginSecurityLog(final HttpServletRequest request,
+                                                    final IdentityProviderConfiguration identityProviderConfiguration,
+                                                    final User user) {
+        try {
+
+            final Host   host  = this.hostWebAPI.getCurrentHost(request);
+            final String env   = this.samlWebUtils.isFrontEndLoginPage(request.getRequestURI()) ? "frontend" : "backend";
+            final String log   = new Date() + ": Successfull SAML login for Site '" + host.getHostname() + "' with IdP " +
+                    "ID: " + identityProviderConfiguration.getId() + " (" + env + ") from " + request.getRemoteAddr() + " for user: " +
+                    user.getEmailAddress();
+
+            // “$TIMEDATE: SAML login success for $host (frontend|backend) from $REQUEST_ADDR for user $username”
+            SecurityLogger.logInfo(SecurityLogger.class, this.getClass() + " - " + log);
+            Logger.info(this, log);
+        } catch (Exception e) {
+
+            Logger.error(this, e.getMessage(), e);
+        }
+    }
+
+    /**
+     * Perform the logic to get or create the user from the SAML and DotCMS If
+     * the SAML_ART_PARAM_KEY, will resolve the Assertion by calling a Resolver
+     * and will create/get/update the user on the dotcms data.
+     *
+     * @param request {@link HttpServletRequest}
+     * @param identityProviderConfiguration {@link IdentityProviderConfiguration}
+     * @return User
+     * @throws IOException
+     * @throws JSONException
+     * @throws DotDataException
+     */
+    public User getUser(final HttpServletRequest request,
+                        final IdentityProviderConfiguration identityProviderConfiguration) {
+
+        User systemUser = null;
```

Comment:
Codacy found an issue: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5707914)